packages/core/src/metrics/MetricsService.ts

```diff
@@ -60,7 +60,7 @@ export class MetricsService {
    * @param type type of the metrics to be published
    * @param metrics metrics instances that should be published
    */
-  public publish(type: string, metrics: Metrics[]): Promise<any> {
+  public async publish(type: string, metrics: Metrics[]): Promise<any> {
```

Comment:
I think we could change the interface so this returns `Promise<MetricsPayload>`

Reply:
It's the MetricsPublisher who determines the return type here really so it should return something Network-related I guess?

Reply:
We removed loggerpublisher from other sdk. Going to do that here as well.